File: my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node


class MyFirstNode(Node):
    def __init__(self):
        super().__init__("py_test")

        self.counter_ = 0

        self.get_logger().info("Hello world")
        self.create_timer(timer_period_sec=1.0, callback=self.timer_callback)

    def timer_callback(self):
        self.get_logger().info("Hello " + str(self.counter_))
        self.counter_ += 1


def main(args=None):
    rclpy.init(args=args)

    node = MyFirstNode()
    rclpy.spin(node=node)

    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include <rclcpp/rclcpp.hpp>

class MyFirstNode : public rclcpp::Node
{
public:
  MyFirstNode()
      : Node("cpp_test")
  {
    RCLCPP_INFO(this->get_logger(), "Hello world");
    pTimer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&MyFirstNode::timerCallback, this));
  }

private:
  void timerCallback()
  {
    RCLCPP_INFO(this->get_logger(), "Hello %d", counter_++);
  }

  rclcpp::TimerBase::SharedPtr pTimer_;
  int counter_ = 0;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  auto pNode = std::make_shared<MyFirstNode>();
  rclcpp::spin(pNode);

  rclcpp::shutdown();
  return 0;
}
